pgsync/sync.py

```diff
@@ -442,14 +442,14 @@ def should_skip_event(self, event: dict) -> bool:
             logger.debug(f"should_skip_event: skipping event, event[\"old\"] and event[\"new\"] is set to None and tg_op is {event['tg_op']}")
             return True
 
-        if event['indices'] is not None:
+        if event['indices'] != type(None) and event['indices'] is not None:
```

Comment:
```python
if event['indices'] is not None:
```
does the same thing, but readable

Reply:
here is json from parse replication slot:

should_skip_event: detected keys with empty values {'scheduled_date', 'qc_results', 'name', 'owner_id', 'description', 'child_node', 'node_housing_id', 'material_need_by', 'survey_type', 'data', 'completed_date'}, but not skipping this event because these keys are not in settings.SKIP_NULL_KEYS which is currently set to ['job_id']
2024-09-04 13:36:45.137:DEBUG:pgsync.sync: about to sync changes from table: mc_hammer.task, changes: [Payload("tg_op: UPDATE, table: task, schema: mc_hammer, old: {}, new: {'id': '9ed52983-988e-44de-b631-31f4994671de', 'job_id': '125158ea-1784-444c-a170-78a4cefd8a09', 'canonical_id': 'TSK1006571323', 'name': None, 'description': None, 'milestone': 'PERMIT', 'type': 'Determine', 'material_need_by': None, 'qc_results': None, 'progress_order': 25, 'scheduled_date': None, 'due_date': "'2024-09-04'", 'owner_id': None, 'completed_date': None, 'status': 'Assigned', 'created_by_id': '0058B0000019lU9QAI', 'created_date': "'2024-09-04", 'last_modified_by_id': '0058B000002rhEkQAI', 'last_modified_date': "'2024-09-04", 'salesforce_id': 'a0pWG000001J6UEYA0', 'salesforce_job_id': 'a0JWG000001Zxsf2AC', 'dlink_checksum': 'bdf0eaba-aab8-00d6-82ee-796ddc1999f5', 'group_id': 'a0Hf400000372V9EAI', 'survey_type': None, 'child_node': None, 'data': None, 'node_housing_id': None, 'origin_system': 'p2'}, xmin: <class 'NoneType'>, indices: <class 'NoneType'>)]

we have indices set into <class 'NoneType'>, it is type(None) = class 'NoneType'

here is event from notification channel:
hould_skip_event: validating event: {'xmin': 730826477, 'new': {'id': '8b2417f8-c663-4dff-814d-0c407dc503e4', 'area_id': 'd5edf2f8-9aa9-4182-9fe6-93
e86e0704a2', 'mode_id': 'e7b0b335-3e10-4f83-a187-c2a8a9b11b7c', 'owner_id': '0058B0000019lRNQAY', 'manager_id': 'cf203f5d-e302-4d02-8614-d733f53e0eee', 'project_salesforce_id': None}, 'old': {
'id': '8b2417f8-c663-4dff-814d-0c407dc503e4', 'area_id': 'd5edf2f8-9aa9-4182-9fe6-93e86e0704a2', 'mode_id': 'e7b0b335-3e10-4f83-a187-c2a8a9b11b7c', 'owner_id': '0058B0000019lRNQAY', 'manager_i
d': 'cf203f5d-e302-4d02-8614-d733f53e0eee', 'project_salesforce_id': None}, 'indices': ['change-orders', 'jobs-without-tasks', 'jobs', 'mc-universe', 'invoices'], 'tg_op': 'UPDATE', 'table': '
job', 'schema': 'mc_hammer', 'changed_fields': None}

from this log we see that changed_fields set to None

Reply:
> _from this log we see that changed_fields set to None_

we are not checking for "`change_fields`" though
we are checking for "indices" 

and this:
```python
if event['indices'] != type(None) and event['indices'] is not None:
```
should do the same thing as this:
```python
if event['indices'] is not None:
```

Reply:
it's not the same. we lost updates from replication slots, we do not checking next condition if event['indices']  = <class 'NoneType'>

Reply:
but you are not checking `NoneType`, you are checking `None`:
```python
event['indices'] != type(None)
```